### PR TITLE
build: modernize ruff config

### PR DESCRIPTION
**Pull request by afuetterer**
## Description

The ruff project recommends using dedicated `[tool.ruff.lint]` sections over `[tool.ruff]`.

This PR also activates the `E` group, see https://docs.astral.sh/ruff/rules/#pycodestyle-e-w

## Types of changes
- [x]  Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes

## Checklist

- [x] I have read the [contributor guide](https://github.com/pangaea-data-publisher/fuji/blob/master/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -148,17 +148,21 @@ markers = [
...
+
+[tool.ruff.lint]
+extend-select = [
   "I",  # isort
   "UP",  # pyupgrade
   "RUF"  # ruff
 ]
-target-version = "py311"
+ignore = [
+  "E722",  # bare-except
```
Comment (afuetterer) [inline pyproject.toml:163]: This should be enabled as well, but I couldn't figure out some of the necessary exceptions to use.

File: fuji_server/helper/metadata_collector_rdf.py [hunk 2]
```diff
@@ -438,9 +438,9 @@ def get_sparqled_metadata(self, g):
                 self.logger.info("FsF-F2-01M : Trying to query generic SPARQL on RDF, found triples: -:" + str(len(g)))
                 r = g.query(Mapper.GENERIC_SPARQL.value)
                 for row in r:
-                    for l, v in row.asdict().items():
-                        if l is not None:
-                            if l in [
+                    for link, v in row.asdict().items():
```
Comment (afuetterer) [inline fuji_server/helper/metadata_collector_rdf.py:441]: Is `link` an appropriate name for `l`? Ruff did not like `l`, due to `ambiguous-variable-name`.

Comment (huberrob) [reply fuji_server/helper/metadata_collector_rdf.py:441]: I think its: v = related_resource; l = relation_type

File: pyproject.toml (hunk 1 above)
Comment (huberrob) [reply pyproject.toml:163]: avoiding bare excepts should be enough  I think it can stay like this

Comment (afuetterer) [reply pyproject.toml:163]: You mean leave it as is?
There are 16 bare excepts in the code base:

https://github.com/search?q=repo%3Apangaea-data-publisher%2Ffuji%20except%3A&type=code

**Comment by afuetterer** [comment]
What do you think?

**Pull request merged**